### PR TITLE
build: Add plugin and Matomo to Dockerfile

### PR DESCRIPTION
**Pull request by abbyoung**
## Description 

* Extends Matomo image in Dockerfile to copy and install plugin
* Updates docker compose file to use env vars for credentials
* Includes LogViewer plugin for testing plugin upload

This example uploads a plugin that lives in the repo, but also includes commented-out instructions for how we could download and install a plugin from the Dockerfile (if that's an option once we purchase the Login SAML plugin).

It's also possible to upload plugins as zip files to the UI, if the setting is enabled in the deployed config.ini.php file.

## Review Notes

To test locally, pull branch and run `docker compose up --build`

Create an .envrc with values for the following variables:
export MARIADB_USER=`<string>`
export MARIADB_PASSWORD=`<string>`
export MARIADB_ROOT_PASSWORD=`<string>`
export MATOMO_USERNAME=`<string>`
export MATOMO_PASSWORD=`<string>`

We'll also need to set these values when deploying.

You should be able to view Matomo at `localhost:8080`

* Click the settings gear in top-right corner
* In the left nav, click Settings > Plugins
* Confirm that LogViewer plugin is in the list, and that you can activate it.
Screen Shot 2022-03-17 at 4 55 40 PM

### CODE REVIEW

**Review by suzubara — COMMENTED** [review]
Followed the instructions & confirmed that I can activate the LogViewer plugin! I definitely want to wait for @esacteksab 's input on the best way to incorporate plugins in our image. Also unrelated to plugins but maybe we can set a few of these initial env vars to use for setup (like first user/password): https://github.com/bitnami/bitnami-docker-matomo#user-and-site-configuration

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# we can install and use unzip here
+# e.g.
+# RUN install_packages unzip
+# RUN curl -o PluginName.zip \
+#     https://plugins.matomo.org/api/2.0/plugins/PluginName/download/latest \
+#     && unzip PluginName.zip \
+#     && rm PluginName.zip \
+#     && mv PluginName /opt/bitnami/matomo/plugins
+
+# # Allow to write in tmp directory
```
Comment (esacteksab) [inline Dockerfile:17]: What's this for?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+version: '3.8'
+
+services:
+  matomo:
+    build: .
+    ports:
+      - '8080:8080'
+      - '443:8443'
```
Comment (esacteksab) [inline docker-compose.yml:8]: Is this meant to be 443 or did you want 8443 to match the convention above?

**Comment by esacteksab** [comment]
Left a few comments and questions, but like I said yesterday, this is how I would have done it. Thank you for doing it! LGTM. 🚢 it!

**Review by esacteksab — APPROVED** [review]

**Comment by esacteksab** [comment]
Is this still necessary? Since we won't be using the SAML plugin and I based my changes off this branch and it will have a docker-compose file in it #6. Can we close this?

**Pull request closed without merging**